### PR TITLE
Extend loglevel logging for the whole project

### LINKED ISSUE #772: Extend logging to whole project
After #763 was merged, let's extend it to the rest of the code base, so we can get rid of `console.log` statements.

### PR DESCRIPTION
**Pull request by jkasun**
Now logging using the loglevel is extend to the whole project instead of `console.log`
Fixes #772

Signed-off-by: Janith Kasun janithisipathana@gmail.com

### CODE REVIEW

File: spec/olm-loader.js [hunk 1]
```diff
@@ -15,10 +15,11 @@ limitations under the License.
 */
 
 // try to load the olm library.
+import logger from '../src/logger';
```
Comment (turt2live) [inline spec/olm-loader.js]: should be above the comment

File: src/models/room.js [hunk 2]
```diff
@@ -555,9 +556,9 @@ Room.prototype.clearLoadedMembersIfNeeded = async function() {
  */
 Room.prototype._cleanupAfterLeaving = function() {
     this.clearLoadedMembersIfNeeded().catch((err) => {
-        console.error(`error after clearing loaded members from ` +
+        logger.error(`error after clearing loaded members from ` +
             `room ${this.roomId} after leaving`);
-        console.dir(err);
+        logger.dir(err);
```
Comment (turt2live) [inline src/models/room.js]: does this actually work?

File: src/realtime-callbacks.js [hunk 3]
```diff
@@ -39,9 +39,10 @@ let _realCallbackKey;
 // each is an object with keys [runAt, func, params, key].
 const _callbackList = [];
 
+import logger from '../src/logger';
+
 // var debuglog = console.log.bind(console);
```
Comment (turt2live) [inline src/realtime-callbacks.js]: should fix the debuglog too, and please add line 44's blank line back

**Comment by jkasun** [comment]
thanks for the review @turt2live. I'll fix these issues and push another commit.

File: src/models/room.js (hunk 2 above)
Comment (jkasun) [reply src/models/room.js]: @turt2live nice catch. Yes this is not working. What can we do about that? Maybe just a log

File: spec/olm-loader.js (hunk 1 above)
Comment (jkasun) [reply spec/olm-loader.js]: This is fixed

File: src/realtime-callbacks.js (hunk 3 above)
Comment (jkasun) [reply src/realtime-callbacks.js]: This is fixed

File: src/models/room.js (hunk 2 above)
Comment (turt2live) [reply src/models/room.js]: just a log is probably fine, yea

Comment (jkasun) [reply src/models/room.js]: Done

**Pull request merged**